storage_manager/app/daf/dataset/DatasetParams.scala

```diff
@@ -91,8 +91,25 @@ object DatasetParams {
     }.map { _.trim }
   }
 
-  private def addParam(key: String, value: Option[String], extraParams: ExtraParams = Map.empty[String, String]) = Success {
+  private def readTheme(catalog: MetaCatalog) = Try {
```

Comment:
What about moving this functions in a dedicated `class` or `object`?
One day I would like to put all the methods which extract info from the `MetaCatalog` in a single place, to avoid replicated code.

Reply:
Where should we put this for the time being though? I wouldn't put it in the `catalog-manager` right now because there's probably more we'd need to do. The `MetaCatalog` class itself (as generated by the plugins) is also a little difficult to work with, so a comprehensive solution would need to start from the `MetaCatalog` class itself. If we intend to use the generated class also in the future, then maybe we can enrich it by pimping it in the `catalog-manager`, but this still opens some broader discussion, for example: do we prefer `Reader`s as a way to extract properties from the `MetaCatalog`, lenses, something more plain or something else entirely?

If you have something in mind, i'm happy to discuss, however i don't think we should do that now with this change, but rather tackle the problem separately.

Reply:
My idea is to keep for the moment those functions inside the service that use them, but in a dedicated place. 
In this way, when in a future we will unify them (in a way still to decide, I like lenses, but is premature now) we will have a single point per service where to look for. 
For example in the `daf-job-ingestion` I've put them in a `DataCatalog` class which simply wraps the `MetaCatalog`

Reply:
Then let's do this if you agree: let's close this pull request when @giux78 also approves, and i'll work on your suggestion immediately afterwards, so as not to hurt these changes from being released.

Reply:
:+1: